### ISSUE TITLE
Use Tensorflow models inside Unity

### ISSUE THREAD

**Issue by PanMig** [Unity-Technologies/ml-agents#2233]
Hello all,

I would like to use some Tensorflow pre-trained models inside Unity. Although they were resources on how to achieve that using the TensorflowSharp plug in, I see that is now deprecated as a solution. How should I approach the issue, should I stick to the old method adding tensorflowSharp and transform models to graphs.byte or there is another newer solution that I missing ?

Thank you in advance

**Comment by xiaomaogy** [comment]
Hi @PanMig Although that is deprecated, the download link is still there. You can still use the older version of ml-agents. Also you might be able to use the Barracuda @mantasp

**Comment by Guidosalimbeni** [comment]
Hi, can you point to the download link? It seems broken... I might be wrong,
A side question. Can I have tensorflowsharp 4.0 (with TF 1.7.1) and the latest version of ML-agents live together in the same UNity project? I understand that UnitySDK of ML-agents comes with a folder TensorFlow and I am not sure if adding the latest UNITY-sdk of ML-agents will cause issue if I have already the tensorflowsharp 4.0 package installed.
thank you for your help

**Comment by PanMig** [comment]
You can find the plug-in here. [https://github.com/salepaun/ML-agents/blob/master/docs/Using-TensorFlow-Sharp-in-Unity.md]

I personally use tensorflow 1.7.1 and the tensorflow-plugin without loading the ML-agents repository on the assets.

**Comment by Guidosalimbeni** [comment]
Dear @PanMig ,
thank you for the link!!
Do you mean that you install the tensorflowsharp plugin first and then you insert the UnitySDK from ML-agents next? 
I probably need to check an older version of the SDK since the latest is using Barracuda... am I right?

**Comment by PanMig** [comment]
@Guidosalimbeni  you DON'T need to insert ML-agents to unity, just use drag and drop the TensorflowSharp plug-in and you will be able to load you own pre-trained models. But beware this is only for loading and using pre-trained tensorflow models and nothing more (based on the title of the issue) .

**Comment by Guidosalimbeni** [comment]
Dear @PanMig ,
yes, sure you are right! Just because in my project I aim to use both (a custom offline CNN for the reward and the PPO for the reinforcement learning) and I forgot about the title of the issue. Sorry. All good!!!
